go/libkb/stream_classifier.go

```diff
@@ -235,7 +235,8 @@ func ClassifyStream(r io.Reader) (sc StreamClassification, out io.Reader, err er
 	peeker := NewStreamPeeker(r)
 	var buf [100]byte
 	var n int
-	if n, err = peeker.Peek(buf[:]); err != nil {
+	// ErrUnexpectedEOF might just mean we read less than 100 bytes
+	if n, err = peeker.Peek(buf[:]); err != nil && !(err == io.ErrUnexpectedEOF && len(buf) > 0) {
```

Comment:
Ah ok that makes sense.  But yeah you should clear out the error for the
case of short messages that are. Amid.

On Monday, February 1, 2016, Steve Sanders <notifications@github.com> wrote:

> In go/libkb/stream_classifier.go
> <https://github.com/keybase/client/pull/1862#discussion_r51518278>:
>
> > @@ -235,7 +235,8 @@ func ClassifyStream(r io.Reader) (sc StreamClassification, out io.Reader, err er
> >  	peeker := NewStreamPeeker(r)
> >  	var buf [100]byte
> >  	var n int
> > -	if n, err = peeker.Peek(buf[:]); err != nil {
> > +	// ErrUnexpectedEOF might just mean we read less than 100 bytes
> > +	if n, err = peeker.Peek(buf[:]); err != nil && !(err == io.ErrUnexpectedEOF && len(buf) > 0) {
>
> UnknownStreamError gets returned anyway in the short message detached
> case, but yeah, I need to do that for the other cases.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1862/files#r51518278>.
>